### PR TITLE
increase the default epochs to 1000 for linear regression

### PR DESCRIPTION
**Pull request by Zhangxunmt**
Signed-off-by: Xun Zhang <xunzh@amazon.com>

### Description
Increase the number of iteration for epochs to improve convergence of linear regression. 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ylwu-amzn — APPROVED** [review]

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public class LinearRegression implements Trainable, Predictable {
     //RMSProp
     private static final double DEFAULT_DECAY_RATE = 0.9;
 
-    private static final int DEFAULT_EPOCHS = 10;
+    private static final int DEFAULT_EPOCHS = 1000;
     private static final int DEFAULT_INTERVAL = -1;
     private static final int DEFAULT_BATCH_SIZE = 1;
```
Comment (ylwu-amzn) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:63]: Can you help fix the hard coded issue in line 203
```
LinearSGDTrainer linearSGDTrainer = new LinearSGDTrainer(objective, optimiser, epochs, DEFAULT_INTERVAL, DEFAULT_BATCH_SIZE, seed);
```

Comment (Zhangxunmt) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:63]: Sure, do you mean DEFAULT_INTERVAL, DEFAULT_BATCH_SIZE, seed these 3 hard coded ones?

Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java:63]: yes

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Review by jngz-es — APPROVED** [review]
